src/js/jsx/search/SearchBar.jsx

```diff
@@ -500,20 +500,29 @@ define(function (require, exports, module) {
 
         render: function () {
             var searchStrings = nls.localize("strings.SEARCH"),
-                noMatchesOption = {
+                filters = this.state.filter,
+                placeholderText,
+                clearInputBtn;
+
+            var flux = this.getFlux(),
+                searchStore = flux.store("search"),
+                registeredSearchModules = searchStore._registeredSearchTypes;
+
+            var lastFilter = filters[filters.length - 1];
+
+            var searchPayload = registeredSearchModules[lastFilter];
+            var noOptionsString = searchPayload ? searchPayload.noOptionsString : null;
```

Comment:
This line and line below can be consolidated into:
`var noOptionsString = searchPayload ? searchPayload.noOptionsString : nls.localize("strings.SEARCH.NO_OPTIONS");`